clip_benchmark/metrics/zeroshot_classification.py

```diff
@@ -44,8 +44,13 @@ def zero_shot_classifier(model, tokenizer, classnames, templates, device, amp=Tr
                 texts = templates[classname]
             else:
                 texts = [template.format(c=classname) for template in templates]
-            texts = tokenizer(texts).to(device)  # tokenize
-            class_embeddings = model.encode_text(texts)
+            texts = tokenizer(texts) # tokenize
+            if isinstance(texts, dict):
+                texts = {k: v.to(device) for k, v in texts.items()}
+                class_embeddings = model.encode_text(**texts)
```

Comment:
might this be needed in https://github.com/LAION-AI/CLIP_benchmark/blob/1597f49d65c15915ad9a36c92c2d18e37ad1ed71/clip_benchmark/metrics/zeroshot_retrieval.py#L54 too ?

could you
* make it so this has the same interface as openclip
* or hide this difference in a wrapper in japanese_clip.py

so we avoid having too many branches in the code ?

Reply:
Yes, you're right. I will fix like so

after
```diff
     for batch_images, batch_texts, inds in tqdm(dataloader):
         batch_images = batch_images.to(device)
         # tokenize all texts in the batch
-        batch_texts_tok = tokenizer([text for i, texts in enumerate(batch_texts) for text in texts]).to(device)
+        batch_texts_tok = tokenizer([text for i, texts in enumerate(batch_texts) for text in texts])
         # store the index of image for each text
         batch_texts_image_index = [ind for ind, texts in zip(inds, batch_texts) for text in texts]
 
         # compute the embedding of images and texts
         with torch.no_grad(), autocast():
             batch_images_emb = F.normalize(model.encode_image(batch_images), dim=-1)
-            batch_texts_emb = F.normalize(model.encode_text(batch_texts_tok), dim=-1)
+            if isinstance(batch_texts_tok, dict):
+                batch_texts_tok = {k: v.to(device) for k, v in batch_texts_tok.items()}
+                batch_texts_emb = F.normalize(model.encode_text(**batch_texts_tok), dim=-1)
+            else:
+                batch_texts_emb = F.normalize(model.encode_text(batch_texts_tok.to(device)), dim=-1)
 
         batch_images_emb_list.append(batch_images_emb.cpu())
         batch_texts_emb_list.append(batch_texts_emb.cpu())
```

Reply:
hi, sorry for the delay

what I meant is it would be great if clip japanese could have the same interface as openclip for model.encode_text and not need this if here

that can be done directly in clip japanese, or a wrapper can be done here

what do you think ?

Reply:
@rom1504 hi, thanks for your comment. I ended up introducing wrappers for ja_clip inside CLIP-benchmark. Then, I don't need to modify any codes inside metrics now. 
https://github.com/LAION-AI/CLIP_benchmark/pull/50/commits/e97e67829ab762de7dd9a7a0212448f9c12729cd